### PR TITLE
Parallelize the converting hash table to two level process after finishing hash agg build

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/8956

Problem Summary:
See https://github.com/pingcap/tiflash/issues/8956.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Aggregator.cpp [hunk 1]
```diff
@@ -2455,14 +2460,55 @@ Block MergingBuckets::getData(size_t concurrency_index)
 {
     assert(concurrency_index < concurrency);
 
-    if (unlikely(data.empty()))
+    if unlikely (data.empty())
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/Aggregator.cpp:2463]: why removing the `()` outside of unlikely? 😂

Comment (gengliqi) [reply dbms/src/Interpreters/Aggregator.cpp:2463]: Just for simplicity😂

File: dbms/src/Operators/AggregateConvergentSourceOp.cpp [hunk 2]
```diff
@@ -31,6 +31,11 @@ AggregateConvergentSourceOp::AggregateConvergentSourceOp(
 
 OperatorStatus AggregateConvergentSourceOp::readImpl(Block & block)
 {
+    if unlikely (!agg_context->convertPendingDataToTwoLevel())
+    {
+        setNotifyFuture(agg_context);
+        return OperatorStatus::WAIT_FOR_NOTIFY;
+    }
```
Comment (SeaRise) [inline dbms/src/Operators/AggregateConvergentSourceOp.cpp:38]: how about adding a new event to convert pending data to two level?

Comment (gengliqi) [reply dbms/src/Operators/AggregateConvergentSourceOp.cpp]: Done.

File: dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp [hunk 3]
```diff
@@ -77,6 +84,39 @@ void PhysicalAggregationBuild::buildPipelineExecGroupImpl(
 EventPtr PhysicalAggregationBuild::doSinkComplete(PipelineExecutorContext & exec_context)
 {
     assert(aggregate_context);
+
+    SCOPE_EXIT({ aggregate_context.reset(); });
+    if (!aggregate_context->hasSpilledData())
```
[the remark this answers is not in the document]
Comment (gengliqi) [reply dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp:89]: Yep.

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp (hunk 3 above)
Comment (windtalker) [inline dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp:89]: This check should at least moved to after `aggregate_context->getAggSpillContext()->finishSpillableStage();` How about move the whole code block after check `AggregateFinalSpillEvent`?

Comment (gengliqi) [reply dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp]: Done.

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**